### PR TITLE
refactor(dht): Simplify promise handling

### PR DESCRIPTION
**Pull request by teogeb**
These changes resolved eslint `promise/always-return` warnings.

## Other changes

- change log level of timeout error and simplified logging in `RoutingSession`
- disabled the warning in two benchmark tests

### CODE REVIEW

**Comment by juslesan** [comment]
Disable virtual time completely in browser tests via webpack to avoid the observed setImmediate issue with virtual time in browsers.

File: packages/dht/src/dht/routing/Router.ts [hunk 1]
```diff
@@ -122,15 +122,20 @@ export class Router implements IRouter {
...
-            raceEvents3<RoutingSessionEvents>(session, ['routingSucceeded', 'partialSuccess', 'routingFailed', 'stopped', 'noCandidatesFound'], 10000)
-                .then(() => {
+            const eventReceived = raceEvents3<RoutingSessionEvents>(
+                session,
+                ['routingSucceeded', 'partialSuccess', 'routingFailed', 'stopped', 'noCandidatesFound'],
+                null
+            )
+            setImmediate(async () => {
+                try {
+                    await withTimeout(eventReceived, 10000)
```
Comment (juslesan) [inline packages/dht/src/dht/routing/Router.ts:135]: Awaiting a promise inside a setImmediate block is a dangarous patter as the promise could be rejected before the await listener is set.

**Comment by teogeb** [comment]
> Disable virtual time completely in browser tests via webpack to avoid the observed setImmediate issue with virtual time in browsers.

Implemented in https://github.com/streamr-dev/network/pull/1850

File: packages/dht/src/dht/routing/Router.ts (hunk 1 above)
Comment (teogeb) [reply packages/dht/src/dht/routing/Router.ts:135]: Using `executeSafePromise` wrapper to underline the fact that this promise can't reject (as it only waits events and never timeouts)

**Review by juslesan — APPROVED** [review]

**Pull request merged**